### DISCUSSION TITLE
Plane and Vec3 Functionality

### DISCUSSION THREAD

**Discussion by KyleN97** [jrouwe/JoltPhysics#1033] [Q&A] [answered]
HI,

I'm working on implementing a pole-vector system within a custom application using Jolt, i was looking and could not find (to my knowledge) the following pre-existing code within Jolt that allows for the following:

-  ClosestPointOnPlane(const JPH::Plane& plane, const JPH::Vec3& point)
- SignedAngle(const JPH::Vec3& from, const JPH::Vec3& to, const JPH::Vec3& axis) ->Calculates the signed angle between vectors from and to in relation to axis

 
 I've implemented my own functionality that allows for this but don't want to have to rewrite code if it already exists! 
 Many Thanks.

**Comment by jrouwe** [comment, marked as answer]
I don't think I have implemented these functions (I often write out these functions where they are needed because it allows me to eliminate duplicate calculations in the context where it is used)
